### PR TITLE
Runs config updated

### PR DESCRIPTION
**Pull request by AhsanInvexion**
Updated the runs config side-tabs as per the new design

### CODE REVIEW

File: src/ui/layouts/runs/RunDetail/Configuration/index.tsx [hunk 1]
```diff
@@ -1,15 +1,97 @@
...
+
 export const Configuration: React.FC<{ runId: TId }> = ({ runId }) => {
   const { run } = useService({ runId });
+  const dispatch = useDispatch();
+
+  const configSchema = `
+from zenml import Client
+
+run = Client().get_pipeline_run(${runId})
+config = run.config
```
Comment (fa9r) [inline src/ui/layouts/runs/RunDetail/Configuration/index.tsx:19]:
```suggestion
from zenml.client import Client

run = Client().get_pipeline_run("${runId}")
config = run.config
```

File: src/ui/components/lineage/sidebarTabsSwitchers/artifactTabSwitcher.tsx [hunk 2]
```diff
@@ -70,6 +78,49 @@ const ArtifactTabHeader = ({
...
+        ></JSONPretty>
+      </Box>
+    </Box>
+  );
+
+  const configSchema = `
+from zenml import Client
+
+artifact = Client().get_artifact(${node?.artifact_store_id})
+loaded_artifact = artifact.load()
```
Comment (fa9r) [inline src/ui/components/lineage/sidebarTabsSwitchers/artifactTabSwitcher.tsx:120]:
```suggestion
from zenml.client import Client

artifact = Client().get_artifact("${node?.artifact_store_id}")
loaded_artifact = artifact.load()
```

File: src/ui/components/lineage/sidebarTabsSwitchers/stepTabSwitcher.tsx [hunk 3]
```diff
@@ -404,6 +491,105 @@ const StepnodeTabHeader: React.FC<any> = ({ node, fetching }) => {
...
+                        <Paragraph>Value</Paragraph>
+                      </td>
+                    </tr>
+                  )}
+                </tbody>
+              </table>
+            </>
+          ) : (
+            ''
+          )}
```
Comment (fa9r) [inline src/ui/components/lineage/sidebarTabsSwitchers/stepTabSwitcher.tsx:592]: Implementation missing?

**Review by fa9r — CHANGES_REQUESTED** [review]
Thanks for the PR! High-level the changes look good, all the pages / tabs are there and the tab order makes sense as well. However, there are still some inconsistencies with the design in some of the pages, see the detailed per-page feedback below. Thanks for your efforts and let me know once it's ready for review again.

### Run Configuration
- **Name field:** on the bottom of the page is useless and should be removed.
- **Failure Hook Source / Success Hook Source:** Should be copyable code path (similar to how you implemented it for the step config page)
- **Get config programmatically:** If I copy and paste it into a code editor, I would like the code to show up as it looks in the dashboard. Currently it is formatted as a single line string `"\nfrom zenml import Client\n\nrun = Client().get_pipeline_run(b0ddd18f-a834-4fb1-9553-e2fec3e92be0)\nconfig = run.config\n\n"`

### Step Details
Still contains very different fields than the design:
- "ID" field is missing
- "Docstring", "Pipelin run ID", "Parent Step IDs", "Cache Enabled", "Inputs", "Outputs" should all be removed
- Order of fields is different from the design

### Step Configuration
Looks perfect 👍  

### Step Metadata
Currently doesn't work.
- If there is no metadata available, the page is currently totally blank. In that case, we should probably display a "No metadata available" text or similar
- If there is metadata available, the "Key" and "Value" headers are shown but the actual metadata is missing. See screenshot below. This is how you reproduce/test this:
    - Go to "Pipelines" and click on a "mlflow_example_pipeline" and go to any of the successful runs
     - Click on "trainer" step in the DAG Visualizer and click on "Metadata" tab.
 
Screenshot 2023-07-07 at 16 28 43

### Artifact Details
- "ID" Field is missing

**Comment by AhsanInvexion** [comment]
@fa9r all of the above comments have been resolved.

except for this point:
"Failure Hook Source / Success Hook Source: Should be copyable code path (similar to how you implemented it for the step config page)"

the API returning failure_hook_source and success_hook_source as null, if these values return as objects then they will be implemented as copyable objects.

File: src/ui/components/lineage/sidebarTabsSwitchers/artifactTabSwitcher.tsx [hunk 4]
```diff
@@ -70,6 +78,49 @@ const ArtifactTabHeader = ({
...
+            fontFamily: 'Rubik',
+          }}
+          data={config}
+        ></JSONPretty>
+      </Box>
+    </Box>
+  );
+
+  const configSchema = `
+from zenml.client import client
```
Comment (fa9r) [inline src/ui/components/lineage/sidebarTabsSwitchers/artifactTabSwitcher.tsx:117]:
```suggestion
from zenml.client import Client
```

**Review by fa9r — CHANGES_REQUESTED** [review]
## Previous Change Requests

The changes I requested last time all look good except for the hook fields as you pointed out. I think it's fine to have them display "Not set" if they are None and only show a copyable code if it is set. However, it is currently inconsistent with the step configs page. Here's how the two pages look like when no source is set:

### Run Config
Screenshot 2023-07-17 at 10 05 20

### Step Config
Screenshot 2023-07-17 at 10 09 55

## New Change Requests
While testing, I also found a few other issues I missed in the first review:
- The Step "Configuration" tab should also have both a "Copy" button that copies the entire config as well as a "Get config programmatically" code snippet, similar to the run configuration page
- "Copy Artifact in Code" is formatted as a single string when I copy-paste it into a code editor instead of being pasted as properly formatted code (Similar to the "Get config programmatically" before)
- The "status" field in the step details is currently misaligned from the rest of the table, see screenshot below
Screenshot 2023-07-17 at 10 09 33

**Review by fa9r — CHANGES_REQUESTED** [review]
Run config looks good now and "Copy artifact in code" also works as expected. 👍 

However, the other two issues I mentioned in my previous review are still not fully resolved:
- The "status" field in the step details is still misaligned from the rest of the table
- The Step "Configuration" tab has a "Copy" button now, but that button copies *both* the entire config and the "Get config programmatically" code snippet at the same time, but these should be separate features, similar to the run config page. E.g., the "Copy" button should only copy the step configuration, and the "Get config programmatically" should be a separate code field on the bottom, similar to how you implemented it for the artifacts.

**Comment by AhsanInvexion** [comment]
@fa9r all above issues has been resolved

**Review by fa9r — APPROVED** [review]
Awesome, looks good to me now, good job @AhsanInvexion  🙌

File: .env
Comment (Cahllagerfeld) [inline .env:1]: Can we please remove the change in the `.env` before merging

**Review by Cahllagerfeld — COMMENTED** [review]
Other than that it looks good to me as well 👍

File: src/ui/components/lineage/index.css [hunk 5]
```diff
@@ -461,11 +455,13 @@
   font-family: 'Rubik';
   font-style: normal;
   font-weight: 400;
-  font-size: 18px;
+  font-size: 12px !important;
```
Comment (Cahllagerfeld) [inline src/ui/components/lineage/index.css:458]: grafik
This change makes the font on all other tabs smaller as well. Is this intended?

Comment (AhsanInvexion) [reply src/ui/components/lineage/index.css:458]: yes. it will make the design consistent for all tabs which use the lineage component

**Pull request merged**